production/loki-mixin/alerts.libsonnet

```diff
@@ -54,16 +54,16 @@
           {
             alert: 'LokiTooManyCompactorsRunning',
             expr: |||
-              sum(loki_boltdb_shipper_compactor_running) by (namespace, cluster) > 1
-            |||,
+              sum(loki_boltdb_shipper_compactor_running) by (namespace, %s) > 1
+            ||| % $._config.per_cluster_label,
             'for': '5m',
             labels: {
               severity: 'warning',
             },
             annotations: {
-              message: |||
+              message: std.strReplace(|||
                 {{ $labels.cluster }} {{ $labels.namespace }} has had {{ printf "%.0f" $value }} compactors running for more than 5m. Only one compactor should run at a time.
-              |||,
+              |||, 'cluster', $._config.per_cluster_label),
```

Comment:
I'm not sure this will work as written, it will look specifically for `cluster` in the string when in reality the values for the labels would be something like `prod-<region>` and `loki-prod-#`, so we would have `prod-<region> loki-prod-# has had <some value> compactors running for more than 5m` as the message.

Reply:
From my tests here, it actually replaces the label correctly in the rule https://github.com/QuentinBisson/loki-upstream/pull/1/files#diff-147d57f875c8b29ffd1bfb7026b4d16da82db9df6b9a2d7dfb1cb82d18fa50a4R36.

I tried to use the % syntax with %s but it did not work due to the `printf "%.0f" $value`

Reply:
ah right, the jsonnet `strReplace` is happening earlier than I thought

is that still not incorrect though? in your code within `config.libsonnet` the `cluster_label` is still `cluster`.

Reply:
I only regenerated the ssd mixins so I changed this file https://github.com/QuentinBisson/loki-upstream/pull/1/files#diff-58c1308d64904785b992854f36800a299e22dbdab03edc9181729b506ad8a3c3R9 :D

Reply:
ah okay! looks correct then